File: app/src/main/java/medlife/com/br/activity/HomeActivity.java
package medlife.com.br.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import android.os.Bundle;
import android.view.View;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.badge.BadgeDrawable;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import medlife.com.br.R;
import medlife.com.br.fragments.*;
import medlife.com.br.helper.UserLocationManager;
import medlife.com.br.helper.UsuarioFirebase;
import medlife.com.br.helper.CartManager;

public class HomeActivity extends AppCompatActivity {

    private BottomNavigationView bottomNavigationView;
    private UserLocationManager locationManager;
    private BadgeDrawable cartBadge;
    private FloatingActionButton fabQuickCart;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);
        
        // Update last login timestamp safely without overwriting other data
        String userId = UsuarioFirebase.getIdUsuario();
        if (userId != null) {
            UsuarioFirebase.atualizarLastLoginSeguro(userId);
        }
        
        // Initialize location manager
        initializeLocationManager();
        
        setupBottomNavigation();
        setupCartBadge();
        setupFloatingActionButton();

        // Set default fragment
        if (savedInstanceState == null) {
            loadFragment(new HomeFragment());
        }
    }
    
    private void initializeLocationManager() {
        locationManager = new UserLocationManager(this);
        
        // Set up location callback for the activity
        locationManager.setLocationCallback(new UserLocationManager.LocationCallback() {
            @Override
            public void onLocationReceived(String address) {
                // Location received, can be used for app-wide location features
                // The HomeFragment will handle its own location display
            }

            @Override
            public void onLocationError(String error) {
                // Handle location errors at activity level if needed
            }
        });
    }

    private void setupBottomNavigation() {
        bottomNavigationView = findViewById(R.id.bottomNavigation);
        bottomNavigationView.setOnItemSelectedListener(item -> {
            int itemId = item.getItemId();
            Fragment fragment = null;
            
            if (itemId == R.id.navigation_home) {
                fragment = new HomeFragment();
            } else if (itemId == R.id.navigation_search) {
                fragment = new SearchFragment();
            } else if (itemId == R.id.navigation_cart) {
                fragment = new CartFragment();
            } else if (itemId == R.id.navigation_favorite) {
                fragment = new FavoriteFragment();
            } else if (itemId == R.id.navigation_profile) {
                fragment = new ProfileFragment();
            }
            
            return loadFragment(fragment);
        });
        
        // Set default selected item
        bottomNavigationView.setSelectedItemId(R.id.navigation_home);
    }

    private boolean loadFragment(Fragment fragment) {
        if (fragment != null) {
            getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.contentFrame, fragment)
                .commit();
            return true;
        }
        return false;
    }
    
    /**
     * Get the location manager instance for use in fragments
     */
    public UserLocationManager getLocationManager() {
        return locationManager;
    }

    private void setupCartBadge() {
        cartBadge = bottomNavigationView.getOrCreateBadge(R.id.navigation_cart);
        updateCartBadge();
    }
    
    public void updateCartBadge() {
        if (cartBadge != null) {
            int cartItemCount = CartManager.getInstance().getCartItems().size();
            if (cartItemCount > 0) {
                cartBadge.setVisible(true);
                cartBadge.setNumber(cartItemCount);
            } else {
                cartBadge.setVisible(false);
            }
        }
        updateFABVisibility();
    }

    private void setupFloatingActionButton() {
        fabQuickCart = findViewById(R.id.fabQuickCart);
        fabQuickCart.setOnClickListener(view -> {
            // Navigate to cart fragment
            loadFragment(new CartFragment());
            bottomNavigationView.setSelectedItemId(R.id.navigation_cart);
        });
        
        // Show FAB only when cart has items
        updateFABVisibility();
    }
    
    private void updateFABVisibility() {
        if (fabQuickCart != null) {
            int cartItemCount = CartManager.getInstance().getCartItems().size();
            fabQuickCart.setVisibility(cartItemCount > 0 ? View.VISIBLE : View.GONE);
        }
    }
}

File: app/src/main/java/medlife/com/br/activity/HomeActivity.kt
package medlife.com.br.activity

import android.os.Bundle
import android.view.View
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.google.android.material.badge.BadgeDrawable
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.android.material.floatingactionbutton.FloatingActionButton
import medlife.com.br.R
import medlife.com.br.fragments.*
import medlife.com.br.helper.CartManager
import medlife.com.br.helper.UserLocationManager
import medlife.com.br.helper.UsuarioFirebase
import medlife.com.br.viewmodel.HomeViewModel

class HomeActivity : AppCompatActivity() {

    private lateinit var bottomNavigationView: BottomNavigationView
    lateinit var locationManager: UserLocationManager
        private set
    private var cartBadge: BadgeDrawable? = null
    private lateinit var fabQuickCart: FloatingActionButton

    private val homeViewModel: HomeViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        val userId = UsuarioFirebase.idUsuario
        userId?.let {
            UsuarioFirebase.atualizarLastLoginSeguro(it)
        }

        initializeLocationManager()
        setupBottomNavigation()
        setupCartBadge()
        setupFloatingActionButton()

        if (savedInstanceState == null) {
            loadFragment(HomeFragment())
        }
    }

    private fun initializeLocationManager() {
        locationManager = UserLocationManager(this)

        locationManager.setLocationCallback(object : UserLocationManager.LocationCallback {
            override fun onLocationReceived(address: String) {
                // Location received
            }

            override fun onLocationError(error: String) {
                // Handle error
            }
        })
    }

    private fun setupBottomNavigation() {
        bottomNavigationView = findViewById(R.id.bottomNavigation)
        bottomNavigationView.setOnItemSelectedListener { item ->
            val fragment: Fragment = when (item.itemId) {
                R.id.navigation_home -> HomeFragment()
                R.id.navigation_search -> SearchFragment()
                R.id.navigation_cart -> CartFragment()
                R.id.navigation_favorite -> FavoriteFragment()
                R.id.navigation_profile -> ProfileFragment()
                else -> return@setOnItemSelectedListener false
            }

            loadFragment(fragment)
        }

        bottomNavigationView.selectedItemId = R.id.navigation_home
    }

    private fun loadFragment(fragment: Fragment): Boolean {
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.contentFrame, fragment)
            .commit()
        return true
    }

    private fun setupCartBadge() {
        cartBadge = bottomNavigationView.getOrCreateBadge(R.id.navigation_cart)
        updateCartBadge()
    }

    fun updateCartBadge() {
        val cartItemCount = homeViewModel.getCartItemCount()
        cartBadge?.let {
            if (cartItemCount > 0) {
                it.isVisible = true
                it.number = cartItemCount
            } else {
                it.isVisible = false
            }
        }
        updateFABVisibility()
    }

    private fun setupFloatingActionButton() {
        fabQuickCart = findViewById(R.id.fabQuickCart)
        fabQuickCart.setOnClickListener {
            loadFragment(CartFragment())
            bottomNavigationView.selectedItemId = R.id.navigation_cart
        }
        updateFABVisibility()
    }

    private fun updateFABVisibility() {
        val cartItemCount = homeViewModel.getCartItemCount()
        if (::fabQuickCart.isInitialized) {
            fabQuickCart.visibility = if (cartItemCount > 0) View.VISIBLE else View.GONE
        }
    }
}
